### ISSUE TITLE
`after:run` is only in running head less mode not on Cypress open [runner]

### ISSUE THREAD

**Issue by bheemaanilconvex** [cypress-io/cypress#26186]
### Current behavior

after:run is only working in headless mode

I'm using experimentalInteractiveRunEvents: true, to enable the after:run actions, after closing cypress open 

### Test code to reproduce

```js       

on("after:run", async (results) => {
                if (results) {
                    try {
                        apiUtils.postToSlackChannel(
                            "Cypress Automation: \n" +
                         
                        );
                    } catch (error) {
                        console.log(error)
                    }
                }
            });
            return config;
        },
        experimentalInteractiveRunEvents: true,
```

### Cypress Version

12.8.1

### Node version

v19.2.0

### Operating System

macOd 13.2.1 (22D68)

**Comment by lmiller1990** [comment]
Hi - you are right, this is not firing when you'd expect. I found it fires after *closing* the browser, the code is here: 

https://github.com/cypress-io/cypress/blob/0a4550ea0c2d50fd1cb2e7f8d8e8bf6c3cb5969d/packages/server/lib/project-base.ts#L264-L290

I suppose this is because there isn't an obvious definition for a "run" in open mode. We also have `atter:spec`, which is *also* not working as expected in open mode. 

I think:

- `after:spec` - when each spec finishes (obviously)
- `after:run` - when the *last* spec finishes (so under experimentalRunAllSpecs, it would only fire once - right at the end)

I started working on this: https://github.com/cypress-io/cypress/pull/26209. It'll need more work, specifically around experiementalRunAllSpecs.

**Comment by lmiller1990** [comment]
It looks like Emily fixed `before:spec` here: https://github.com/cypress-io/cypress/commit/c8b1b303234f8835dadbfc8aeea840f67512ea54. The pattern is likely the same. We might also want to verify `before:run` works as expected in open mode.

**Comment by emilyrohrbough** [comment]
Duplicate of https://github.com/cypress-io/cypress/issues/23627

**Comment by lmiller1990** [comment]
Ok, let's close and track there.
